x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingDeciderService.java

```diff
@@ -367,9 +367,9 @@ public AutoscalingDeciderResult scale(Settings configuration, AutoscalingDecider
         if (isMaster == false) {
```

Comment:
Not relevant to this PR but this is one scary long method. Should we raise an issue on looking to refactor this? Or is it just not possible to do much better here due to the nature of what it's doing?

Reply:
It's probably possible to reduce the size.

I still need to iterate on this PR first though.  There's at least one more bug, maybe two, and these need fixing before 8.3 feature freeze.

Reply:
Certainly, I wasn't suggesting to do anything as part of this PR.